language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -850,8 +853,8 @@ infixr 6 <.> -- This is the same fixity as '<>'.
 -- TokenInterface<Asset>`. If the template implements a second `Other` interface, the type `ContractId
 -- Token` needs to be replaced with `ContractId (TokenInterface<Asset> & OtherInterface<Asset>)` and
 -- `ContractId Other` with `ContractId (TokenInterface<Asset> & OtherInterface<Asset>)`.
-genType :: TypeRef -> Maybe (Set.Set T.Text, T.Text) -> (T.Text, T.Text)
-genType (TypeRef curModName t) mbSubst = go t
+genType :: TypeRef -> Maybe (Set.Set TsTypeRef, T.Text) -> GenType
```

Comment:
As the `TConApp` branch reveals, the meaning of this return type isn't the same as that of `genTypeCon`, hence the use of separate types for that function and `genType`.

Reply:
When the type has parameters, the type-ref from `genTypeCon` will be `T` whereas the one from `genType` will be `T<A, B, ...>`. Likewise, the serializer-ref will be `Tvalue` versus `Tvalue(evA, evB, ...)`.

Reply:
That makes sense but the names don’t really help here. maybe consider using a `TsTypeRef` newtype for the references produced by genType and `TsTypeConRef` newtype for `genTypeCon` (similarly for the serializers). Currently you have a record field called `typeRef` which is different from `TsTypeRef`.

Reply:
How about 7feff7da3d6?

Reply:
:+1: